erpnext/hr/doctype/salary_slip/salary_slip.py

```diff
@@ -452,10 +452,10 @@ def calculate_net_pay(self):
 
 		self.set_loan_repayment()
 
-		self.net_pay = flt(self.gross_pay) - (flt(self.total_deduction) + flt(self.total_loan_repayment))
+		self.net_pay = (flt(self.gross_pay) - (flt(self.total_deduction) + flt(self.total_loan_repayment))) * flt(self.payment_days / self.total_working_days)
```

Comment:
Might need a check for total working days being greater than zero, to avoid a zero-division error.

Reply:
total working days is always going to be number of days in a month and can never be zero

Reply:
Ah, looks good then.